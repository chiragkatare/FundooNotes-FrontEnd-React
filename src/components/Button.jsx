import React, { Component } from "react";
import Button from '@material-ui/core/Button';

