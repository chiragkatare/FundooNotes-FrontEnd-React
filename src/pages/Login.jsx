import React, { Component } from "react";
import Input from "../components/Input"
import { Card, Typography, Button, CardContent } from '@material-ui/core/';
import UserService from "../services/UserService";
import { Redirect } from "react-router-dom"
// eslint-disable-next-line
import GoogleLogin from '../components/GoogleLogin';
import FacebookLogin from '../components/FacebookLogin';
import GoogleTest from '../services/GoogleLoginService';

var googleLoginService = new GoogleTest();
var userService = new UserService();

class Login extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      password: '',
      isValid: '',
      error: '',
      isLoggedIn: false,
    }
    this.getDataFromInput = this.getDataFromInput.bind(this);
    this.validate = this.validate.bind(this);
    this.handleClick = this.handleClick.bind(this);

  }

  /**
   * function to handle the input validations of the page
   * 
   * @param {event} event 
   */
  validate(event) {
    var error = {
      email: '',
      password: '',
    };
    var isvalid = true;
    // if(event.target.name==='firstname'){

    if (this.state.email.length < 1) {
      error.email = 'Email is required'
      isvalid = false;
    }
    //}
    //else if(event.target.name==='password'){
    if (this.state.password.length < 1) {
      error.password = 'Password cannot be empty'
      isvalid = false;
    }
    this.setState({
      error: error,
    })
    return isvalid;
  }

  /**
   * function to get data from the object
   * 
   * @param {var} data 
   */
  getDataFromInput(event, data) {
    this.setState({
      [event.target.name]: data
    })
  }

  /***
   * function to handle click of the button of login
   * 
   * @param {event} event 
   */
  handleClick() {
    if (this.validate()) {
      let data = {
        email: this.state.email,
        password: this.state.password,
      }
      userService.login(data).then(response => {
        console.log("asasa", response);
        console.log(this.state);

        if (response.status === 200) {
          this.setState({
            isLoggedIn: true,
          });
        }
        else if (response.status === 204) {
          this.setState({
            error: {
              email: 'invalid credentials',
            }
          })
        }
        else if (response.status === 211) {
          localStorage.removeItem('fundootoken');
          this.props.history.push('/notverified');
        }


      }
      ).catch(error => {
        //console.log("401",error);

      });
    }
  }

  googleTest = () => {
    googleLoginService.googlelogin();
  }

  changeLoginStatus = () => {
    this.setState({
      isLoggedIn: true,
    });
  }



  /**
   * render function to render on html page
   */
  render() {

    /**
     * redirect user if already logged in
     */
    if ((localStorage.getItem('fundootoken')) !== null || this.state.isLoggedIn === true) {
      return (<Redirect to='/dashboard'></Redirect>);
    }

    return (

      <div>
        <script src="https://apis.google.com/js/platform.js" async defer></script>
        <Card id='logincard'>
          <CardContent>
            <div>
              <Typography variant="h5" component="h2" color='primary' id='login-text'>
                Login
        </Typography>
              <div>
                <Input name='email' type={'Email'} placeholder={'Enter Your Email'} label={'Email'} onChange={this.getDataFromInput} />
                <div className='error'>{this.state.error.email}</div>
              </div>
              <div>
                <Input name='password' type={'Password'} placeholder={'Enter PassWord'} label={'PassWord'} onChange={this.getDataFromInput} />
                <div className='error'>{this.state.error.password}</div>
              </div>
              <div id='login-btn-div'>
                <Button variant="contained" color="primary" type='submit' onClick={this.handleClick} className='login-btn'>
                  Login
                </Button>
              </div>
            </div>
            <div >
              <span className='below-txt' >
                <span><a href="/forgetpassword">Forgot Password</a></span>
              </span>
              <span >
                <Typography >New User,<a href="/register">SignUp</a></Typography>
              </span>
            </div>
            <div className='social-login' >
              {/* <Button variant="contained" color="primary" type='submit' onClick={this.googleTest} className='login-btn'>
                  Google Test
                </Button> */}
              <GoogleLogin
                changeLoginStatus={this.changeLoginStatus} />
              <FacebookLogin
                changeLoginStatus={this.changeLoginStatus} />
            </div>
          </CardContent>
        </Card>
      </div>

    );
  }

}

export default Login;