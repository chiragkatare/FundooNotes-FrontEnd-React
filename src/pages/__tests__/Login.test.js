import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import Login from '../Login';
const mockFn = jest.fn();
import Button from '@material-ui/core/Button'

// configure({ adapter: new Adapter() });

describe('Login', () => {

  it('renders a email input', () => {
    const componentInput = shallow(<Login />);
    expect(componentInput.find('#email').length).toEqual(1);
  })

  it('renders a password input', () => {
    const componentInput = shallow(<Login />);
    expect(componentInput.find('#password').length).toEqual(1);
  });

  it('should render correctly in "debug" mode', () => {
    const component = shallow(<Login debug />);
    // const component = mount(<Login debug />);
    expect(component).toMatchSnapshot();
  });



  it('should not login with empty email', () => {
    const component = shallow(<Login />);
    const loginButton = component.find('#loginbtn')
    loginButton.simulate('click');
    const errordiv = component.find('div.error').at(0)
    const text = errordiv.text()
    expect(text).toEqual('Email is required')
  });


  it('should not login with empty password', () => {
    const component = shallow(<Login />);
    const loginButton = component.find('#loginbtn')
    loginButton.simulate('click');
    const errordiv = component.find('div.error').at(1)
    const text = errordiv.text()
    expect(text).toEqual('Password cannot be empty')

  });

  it('email input respond to change event and change state of the Login Component', () => {

    const componentInput = shallow(<Login />);
    const emailTest = componentInput.find('#email');

    emailTest.props().onChange({
        target: {
            name: 'email',
            value: 'abcd@gmail.com'
        }
    });

    expect(componentInput.state('email')).toEqual('abcd@gmail.com');
})

it('password input Should respond to change event and change state of the Login Component', () => {

  const componentInput = shallow(<Login />);
  const emailTest = componentInput.find('#password');

  emailTest.props().onChange({
      target: {
          name: 'password',
          value: 'abcd@123'
      }
  });

  expect(componentInput.state('password')).toEqual('abcd@123');
})

describe('Test Button component', () => {
  it('Test click event', () => {
      const mockCallBack = jest.fn();
      const button = shallow((<Button onClick={mockCallBack}>Ok!</Button>));
      button.simulate('click');
      expect(mockCallBack.mock.calls.length).toEqual(1);
  });
});


});