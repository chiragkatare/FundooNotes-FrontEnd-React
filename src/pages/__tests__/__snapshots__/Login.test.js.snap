// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render correctly in "debug" mode 1`] = `
<div>
  <script
    async={true}
    defer={true}
    src="https://apis.google.com/js/platform.js"
  />
  <WithStyles(Card)
    id="logincard"
  >
    <WithStyles(CardContent)>
      <div>
        <WithStyles(Typography)
          color="primary"
          component="h2"
          id="login-text"
          variant="h5"
        >
          Login
        </WithStyles(Typography)>
        <div>
          <Input
            id="email"
            label="Email"
            name="email"
            onChange={[Function]}
            placeholder="Enter Your Email"
            type="Email"
          />
          <div
            className="error"
          />
        </div>
        <div>
          <Input
            id="password"
            label="PassWord"
            name="password"
            onChange={[Function]}
            placeholder="Enter PassWord"
            type="Password"
          />
          <div
            className="error"
          />
        </div>
        <div
          id="login-btn-div"
        >
          <WithStyles(Button)
            className="login-btn"
            color="primary"
            id="loginbtn"
            onClick={[Function]}
            type="submit"
            variant="contained"
          >
            Login
          </WithStyles(Button)>
        </div>
      </div>
      <div>
        <span
          className="below-txt"
        >
          <span>
            <a
              href="/forgetpassword"
            >
              Forgot Password
            </a>
          </span>
        </span>
        <span>
          <WithStyles(Typography)>
            New User,
            <a
              href="/register"
            >
              SignUp
            </a>
          </WithStyles(Typography)>
        </span>
      </div>
      <div
        className="social-login"
      >
        <GoogleLogin
          changeLoginStatus={[Function]}
        />
        <FacebookLogin
          changeLoginStatus={[Function]}
        />
      </div>
    </WithStyles(CardContent)>
  </WithStyles(Card)>
</div>
`;
